packages/protocol/contracts/ERC1724/base/ZkAssetMintableBase.sol

```diff
@@ -158,5 +147,35 @@ contract ZkAssetMintableBase is ZkAssetOwnableBase {
             emit RedeemTokens(publicOwner, uint256(publicValue));
         }
     }
+
+    /**
+    * @dev called when a mintable and convertible asset wants to perform an
+            action which puts the zero-knowledge and public
+            balance out of balance. For example, if minting in zero-knowledge, some
+            public tokens need to be added to the pool
+            managed by ACE, otherwise any private->public conversion runs the risk of not
+            having any public tokens to send.
+    *
+    * @param _value the value to be added
+    */
+    function supplementTokens(uint256 _value) internal {
```

Comment:
I like this, redefining the supplementTokens() related logic into the same function